ce/ce/artifacts/activation.ts

```diff
@@ -303,9 +299,19 @@ export class Activation {
     return v ? await this.resolveAndVerify(v) : undefined;
   }
 
-  get propertyCount() {
-    return this.#properties.size;
+  msBuildProcessPropertyValue(value: string, targetFolder: Uri) {
+    // note that this is intended to be consistent with NuGet's handling:
+    // https://docs.microsoft.com/en-us/nuget/reference/nuspec#replacement-tokens
+    const initialLocal = targetFolder.fsPath;
+    const endsWithSlash = initialLocal.endsWith('\\') || initialLocal.endsWith('/');
+    const root = endsWithSlash ? initialLocal.substring(0, initialLocal.length - 1) : initialLocal;
+    return value.replaceAll('$root$', root);
```

Comment:
While this does happen to match NuGet's handling, I don't think that's necessarily the best model to apply here -- especially because there's no documented escape method. NuGet's .nuspec file is also XML, not JSON and so perhaps had different constraints.

I'd prefer to match existing vcpkg templating capabilities (such as `api_stable_format`), which use curly-braces (and have clear escape patterns -- '{{' and '}}').

My immediate suggestion would be to `.replaceAll('{root}', root)`, with open bugs to:
1. Add escaping for `{{` and `}}`.
2. Add error cases for unmatched values.

Reply:
This is what Olga asked for; please work it out with her.

Reply:
Robert, we don't need any escape mechanism here as $root$ cannot appear in msbuild property value - it would be invalid and would need to be escaped (for msbuild to be valid). Also making it similar to Nuget is beneficial as people who will create artifacts with msbuild properties will likely be familiar with Nuget.

Reply:
actually, I am wrong here - it can be in a property value and it is a valid msbuild. It is very unlikely though that it would be used there. We use replacenemt tokens in other places (like project templates), let me check how they look like.

Reply:
yes, the project templates also use $value$ for replacement tokens.
```
 <RootNamespace>$safeprojectname$</RootNamespace>
 <DefaultLanguage>$currentuiculturename$</DefaultLanguage>
```
so I don't think we need to complicate it.

Reply:
Because this is a user-facing vcpkg artifacts feature, we should more strongly prefer consistency across the vcpkg product and not prefer consistency with VS project template authoring.

Is there a reason why `{root}`, with `{{` and `}}` escapes, is not sufficient here?

Reply:
It is sufficient and I see your point about consistency. 

My comment about project templates was just for illustration that using $root$ without any escaping is good enough.
The escape support complicates the replacement, but with '{root}' it will be needed.

Reply:
And to clarify my prev comment - I am fine with the '{root}' and escape.

Reply:
I believe https://github.com/microsoft/vcpkg-tool/pull/672/commits/5df752402d223e45ff2dbce57dbab4608d4c488d implements this?